### PR TITLE
add has_anonymous_username and has_anonymous_display_name to user account api

### LINKED ISSUE #5838: Expose `has_anonymous_username` for User
If a user hasn't set their profile name, they have an ugly "anonymous-127537" type username, which will appear in the top of the page's UI on the frontend and across the app. This isn't a great UX, it would be better to expose simply "Anonymous User" or something similar based on a flag set saying they haven't set their username yet. If they haven't set it we could even use their email.

Apparently there's a `has_anonymous_username` method on the User profile we could use. This is problematic as if a user actually has a username that matches it we'd still display their email or "Anonymous User", translated.

Ultimately it would be nice if the first user `save()` done with an anonymous name had a flag explicitly set like `has_set_username` to be `false`. And then once the username was changed it was set to `true`. I think a migration that set all users with `has_anonymous_username() === true` as having that flag is fine but going forward I think that should be an explicit setting.

Then it would be good to expose in the API so we can decide how to expose a username in the frontend.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #5838

### CODE REVIEW

File: docs/topics/api/accounts.rst [hunk 1]
```diff
@@ -37,6 +37,8 @@ A developer is defined as a user who is listed as a developer or owner of one or
     :>json string|null picture_type: the image type (only 'image/png' is supported) if a user defined photo has been provided, or none if no photo has been provided.
     :>json boolean is_addon_developer: The user has developed and listed add-ons on this website.
     :>json boolean is_artist: The user has developed and listed themes on this website.
+    :>json boolean has_anonymous_display_name: The user hasn't chosen a name or username.
```
Comment (tofumatt) [inline docs/topics/api/accounts.rst]: I would rephrase this to "The user has chosen neither a name nor a username." I know it's more verbose but it should be clear it's about having chosen neither, not having chosen neither or only one.

Comment (eviljeff) [reply docs/topics/api/accounts.rst]: https://github.com/mozilla/addons-server/pull/6426/commits/5478817743bc9827f0036a08da376040c2eb1def
